x-pack/platform/plugins/shared/task_manager/server/task_claimers/lib/task_selector_by_capacity.ts

```diff
@@ -8,23 +8,34 @@
 import type { ConcreteTaskInstance } from '../../task';
 import type { TaskClaimingBatches } from '../../queries/task_claiming';
 import { isLimited } from '../../queries/task_claiming';
+import { sharedConcurrencyTaskTypes, type TaskTypeDictionary } from '../../task_type_dictionary';
 
+interface SelectTasksByCapacityOpts {
+  definitions: TaskTypeDictionary;
+  tasks: ConcreteTaskInstance[];
+  batches: TaskClaimingBatches;
+}
 // given a list of tasks and capacity info, select the tasks that meet capacity
-export function selectTasksByCapacity(
-  tasks: ConcreteTaskInstance[],
-  batches: TaskClaimingBatches
-): ConcreteTaskInstance[] {
+export function selectTasksByCapacity({
+  definitions,
+  tasks,
+  batches,
+}: SelectTasksByCapacityOpts): ConcreteTaskInstance[] {
   // create a map of task type - concurrency
   const limitedBatches = batches.filter(isLimited);
-  const limitedMap = new Map<string, number>();
+  const limitedMap = new Map<string, number | null>();
   for (const limitedBatch of limitedBatches) {
-    const { tasksTypes, concurrency } = limitedBatch;
-    limitedMap.set(tasksTypes, concurrency);
+    const { tasksTypes: taskType } = limitedBatch;
```

Comment:
I made this change because the `concurrency` value here is actually an `enum` indicating whether the task type was `Limited` or `Unlimited`. I made the change to look up the concurrency from the task definition. 

This currently works because we only have 1 limited concurrency task and that concurrency happens to be 1 but would break if we had a task with `maxConcurrency` greater than 1.

Reply:
Correct. On main we only have one task type (report) with a maxConcurrency value set and it is set to 1. This happens to match exactly with the enum for `TaskConcurrency.Limited` which resolves to 1.

If we had a task type with maxConcurrency set to 2, the previous code would still set concurrency to 1 because in this case, concurrency is the enum, not the `maxConcurrency` value.